test/conformance/ptp/ptp.go

```diff
@@ -943,3 +980,94 @@ func ptpEventEnabled() bool {
 	}
 	return ptpConfig.Spec.EventConfig.EnableEventPublisher
 }
+
+func getNICInfo(pod v1core.Pod) map[string]string {
+	var ptpSupportedInterfaces []string = getPtpMasterSlaveAttachedInterfaces(pod)
+	var stdout bytes.Buffer
+
+	var ptpInterfaceNicMapping = make(map[string]string)
+
+	for _, interf := range ptpSupportedInterfaces {
+		PCIAddr := ""
+		var err error
+
+		Eventually(func() error {
+			stdout, err = pods.ExecCommand(client.Client, pod, PtpContainerName, []string{"readlink", "-f", fmt.Sprintf("/sys/class/net/%s", interf)})
+			if err != nil {
+				return err
+			}
+
+			if stdout.String() == "" {
+				return fmt.Errorf("empty response from pod retrying")
+			}
+
+			pathSegments := strings.Split(stdout.String(), "/")
+			PCIAddr = pathSegments[5] // 0000:19:00.5
+			return nil
+		}, 3*time.Minute, 5*time.Second).Should(BeNil())
+
+		if PCIAddr == "" {
+			continue
+		}
+
+		ptpInterfaceNicMapping[interf] = PCIAddr
+	}
+
+	return ptpInterfaceNicMapping
+}
+
+func getPtpOperatorVersion() (string, error) {
+	var ptpOperatorVersion string
+
+	deploy, err := client.Client.AppsV1Interface.Deployments(PtpLinuxDaemonNamespace).Get(context.TODO(), PtpOperatorDeploymentName, metav1.GetOptions{})
+
+	if err != nil {
+		logrus.Infof("PTP Operator version is not found: %v", err)
+		return "", err
+	}
+
+	envs := deploy.Spec.Template.Spec.Containers[0].Env
+	for _, env := range envs {
+
+		if env.Name == "RELEASE_VERSION" {
+			ptpOperatorVersion = env.Value
+			ptpOperatorVersion = ptpOperatorVersion[1:]
+		}
+	}
+
+	fmt.Printf("PTP operator version is %v\n", ptpOperatorVersion)
+
+	return ptpOperatorVersion, err
+}
+
+func getOCPVersion() (string, error) {
+	ocpClient := client.Client.OcpClient
+	clusterOperator, err := ocpClient.ClusterOperators().Get(context.TODO(), "openshift-apiserver", metav1.GetOptions{})
+
+	var ocpVersion string
+	if err != nil {
+		switch {
+		case kerrors.IsForbidden(err), kerrors.IsNotFound(err):
+			logrus.Infof("OpenShift Version not found (must be logged in to cluster as admin): %v", err)
+			err = nil
+		}
+	}
+	if clusterOperator != nil {
+		for _, ver := range clusterOperator.Status.Versions {
+			if ver.Name == "openshift-apiserver" {
+				ocpVersion = ver.Version
+				break
+			}
+		}
+	}
+	fmt.Printf("OCP Version is %v\n", ocpVersion)
+
+	return ocpVersion, err
+}
+
+func printInterface(config ptpv1.PtpConfig, interfaceDetailsMap map[string]string) {
+	for _, profile := range config.Spec.Profile {
```

Comment:
since profile is a pointer , you could  get nil pointer exception

Reply:
`Profile` is giving an array - so, NPE does not occur here, imo. Still I added this condition `if (ptpv1.PtpProfile{}) == profile {
			continue
		}`

Let me know what you think.

Reply:
I checked the if profile is empty, NPE cannot happen here, imo, as this is not a pointer.